contracts/src/v0.8/dev/keeper2_0/KeeperRegistrar2_0.sol

```diff
@@ -412,12 +400,13 @@ contract KeeperRegistrar2_0 is TypeAndVersionInterface, ConfirmedOwner, ERC677Re
    * @param expected address that should match the actual sender address
    * @param data bytes
    */
-  modifier isActualSender(address expected, bytes memory data) {
-    address actual;
-    assembly {
-      actual := mload(add(data, 292))
-    }
-    if (expected != actual) {
+  modifier isActualSender(address expected, bytes calldata data) {
+    // decode register function arguments to get actual sender
+    (, , , , , , , address sender) = abi.decode(
+      data[4:],
```

Comment:
I believe `4` is to remove the function signature? In function above `permittedFunctionsForLINK` we add 32 bytes to data and then get selector. I'm not sure how both are working at the same time

Reply:
add(X, 32) is for a `memory` string so the first 32 bytes contains the length and data follows. So add(X, 32) skips that and points to the data.
vs
The first four bytes of a contract call(`calldata`) is the function selector